### PR TITLE
Show enum variant on Self qualified paths

### LINKED ISSUE #6550: Variants auto completion when enums are references
lib_rs_—_reign

If `a` is of type `A`:

* When I type `A`, I get `A` and `A::Foo`
* When I type `F`, I get `A::Foo`

But as you can see from above image, if `a` is of type `&A` (same issue with `&mut A`):

* When I type `A`, I get `A`
* When I type `F`, I get neither

### PR DESCRIPTION
**Pull request by Veykril**
Fixes first part of #6549
Fixes #6550

### CODE REVIEW

File: crates/completion/src/completions/unqualified_path.rs [hunk 1]
```diff
@@ -721,6 +726,26 @@ fn main() {
...
+            r#"
+enum Foo { Bar, Baz, Quux }
+
+fn main() {
+    let foo = Foo::Quux;
+    match &foo { Qu<|> }
+}
+"#,
+            expect![[r#"
+                ev Foo::Bar  ()
```
Comment (lnicola) [inline crates/completion/src/completions/unqualified_path.rs:741]: Should we really suggest `Foo::Bar` and `Foo::Baz` after `Qu`?

Comment (Veykril) [reply crates/completion/src/completions/unqualified_path.rs:741]: I think we make it the clients job to filter out the completions? I'm not sure but most(if not all) completions ignore the already typed parts from what I've seen.

Comment (matklad) [reply crates/completion/src/completions/unqualified_path.rs:741]: yup, we should send *all* completion variants. 

Well, actually, now that we want to implement sorting ourselves, I *think* we should implement sorting ourselves (as a generic filter higher in the stack, not in each specific completion contributor), if LSP allows for that.

**Pull request merged**